site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -1676,6 +1676,10 @@ public function showGrading(
             // Get the graded gradeable for the $from user
             $id_from_anon = $this->core->getQueries()->getSubmitterIdFromAnonId($from, $gradeable_id);
             if ($blind_grading !== "unblind" || $anon_mode) {
+                if ($id_from_anon === null) {
+                    header('Location: /home');
+                    exit;
```

Comment:
We do not want to abruptly exit, or set headers like this.  It would be better to use `return new RedirectResponse('/home');` instead.  That said, is this query parameter mandatory?  If so, should it simply be added to the route?

Reply:
"The submitter id " query is mandatory for the specific route to be able to grade that specific student. I think it should return to the home page, because there isn't a default submitter it should go to if no "submitter id" is provided.

Reply:
Yes, we should update the route then.